mlflow/genai/judges/base.py

```diff
@@ -0,0 +1,127 @@
+    model: str = Field(..., description="LLM model identifier")
+
+    _examples: list[dict[str, Any]] = PrivateAttr(default_factory=list)
+    _prompt_template: str | None = PrivateAttr(default=None)
+
+    def __init__(
+        self,
+        name: str,
+        instructions: str,
+        model: str,
+        examples: list[dict[str, Any]] | None = None,
+        aggregations: list[Any] | None = None,
+        **kwargs,
+    ):
+        """
+        Initialize a Judge.
+
+        Args:
+            name: Unique identifier for the judge
+            instructions: Human-readable instructions defining what the judge evaluates
+            model: LLM model identifier (LiteLLM format, e.g., "openai/gpt-4o-mini")
+            examples: Optional few-shot examples for alignment
+            aggregations: Optional aggregation functions for the judge's output
+        """
+        super().__init__(
+            name=name, instructions=instructions, model=model, aggregations=aggregations, **kwargs
+        )
+        self._examples = examples or []
+        self._prompt_template = None
+
+    def align(self, traces: list[Trace]) -> "Judge":
+        """
+        Create an aligned version of this judge based on labeled traces.
+
+        Args:
+            traces: List of traces with expectations (human labels)
+
+        Returns:
+            A new Judge instance with improved alignment
+        """
+        # TODO: Implement alignment logic
+        raise NotImplementedError(
+            "Judge alignment is not yet implemented. This will be available in a future release."
+        )
+
+    @property
+    def description(self) -> str:
+        """Human-readable description of the judge."""
+        return self.instructions
+
+    def __call__(
+        self,
+        *,
+        inputs: dict[str, Any] | None = None,
+        outputs: dict[str, Any] | None = None,
+        expectations: dict[str, Any] | None = None,
+        trace: Trace | None = None,
+    ) -> Assessment | Feedback:
+        """
+        Evaluate inputs/outputs or a trace.
+
+        Args:
+            inputs: Input values for evaluation
+            outputs: Output values for evaluation
+            expectations: Expected values (optional)
+            trace: MLflow trace (alternative to inputs/outputs)
+
+        Returns:
+            Assessment with the judge's evaluation
+        """
+        from mlflow.genai.judges.utils import invoke_judge_model
+
+        prompt = self._prepare_prompt(inputs, outputs, expectations, trace)
+
+        result = invoke_judge_model(
+            model=self.model,
+            prompt=prompt,
+            name=self.name,
+        )
+
+        if isinstance(result, (str, bool, int, float)):
+            result = Feedback(
+                name=self.name,
+                value=result,
+                rationale=f"Evaluated by {self.name} judge using {self.model}",
+            )
+
+        return result
```

Comment:
Let's leave `__call__` abstract, as defined in `Scorer`. We'll fill it in for individual judge implementations (see https://github.com/mlflow/mlflow/pull/17328/files#r2289473377)

Reply:
Sounds good! Having it directly in here on effectively an ABC is pointless